test/groovy/TransportRequestCreateTest.groovy

```diff
@@ -0,0 +1,96 @@
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.ExpectedException
+import org.junit.rules.RuleChain
+
+import util.BasePiperTest
+import util.JenkinsStepRule
+import util.JenkinsLoggingRule
+import util.Rules
+
+import hudson.AbortException
+
+
+public class TransportRequestCreateTest extends BasePiperTest {
+
+    private ExpectedException thrown = new ExpectedException()
+    private JenkinsStepRule jsr = new JenkinsStepRule(this)
+    private JenkinsLoggingRule jlr = new JenkinsLoggingRule(this)
+
+    @Rule
+    public RuleChain ruleChain = Rules.getCommonRules(this)
+        .around(thrown)
+        .around(jsr)
+        .around(jlr)
+
+    @Before
+    public void setup() {
+
+        helper.registerAllowedMethod('usernamePassword', [Map.class], {m -> return m})
+
+        helper.registerAllowedMethod('withCredentials', [List, Closure], { l, c ->
+
+            credentialsId = l[0].credentialsId
+            binding.setProperty('username', 'anonymous')
+            binding.setProperty('password', '********')
+            try {
+                c()
+            } finally {
+                binding.setProperty('username', null)
+                binding.setProperty('password', null)
+            }
+         })
+
+        helper.registerAllowedMethod('sh', [Map], { Map m -> return 0 })
+
+        nullScript.commonPipelineEnvironment.configuration = [steps:
+                                     [transportRequestCreate:
+                                         [
+                                          cmCredentialsId: 'CM',
+                                          cmEndpoint: 'https://example.org/cm'
+                                         ]
+                                     ]
+                                 ]
+    }
+
+    @Test
+    public void changeIdNotProvidedTest() {
+
+        thrown.expect(AbortException)
+        thrown.expectMessage("Change id not provided (parameter: 'changeId').")
+
+        jsr.step.call(script: nullScript, developmentSystemId: '001')
+    }
+
+    @Test
+    public void developmentSystemIdNotProvidedTest() {
+
+        thrown.expect(AbortException)
+        thrown.expectMessage("Development system id not provided (parameter: 'developmentSystemId').")
+
+        jsr.step.call(script: nullScript, changeId: '001')
+    }
+
+    @Test
+    public void createTransportRequestFailureTest() {
+
+        helper.registerAllowedMethod('sh', [Map], { Map m -> throw new AbortException('Exception message.') })
+
+        thrown.expect(AbortException)
+        thrown.expectMessage("Cannot create a transport request for change id '001'. Exception message.")
+
+        jsr.step.call(script: nullScript, changeId: '001', developmentSystemId: '001')
+    }
+
+    @Test
+    public void createTransportRequestSuccessTest() {
+
+        helper.registerAllowedMethod('sh', [Map], { Map m -> return '001' })
+
+        jsr.step.call(script: nullScript, changeId: '001', developmentSystemId: '001')
+
```

Comment:
It is IMO fine to assert also log messages in case they are important for troubleshooting. But basically I would expect a check for either (1) the command line or (2) for the call to ChangeManagement#createTransportRequest.

I personally prefer and recommand option (2). Just check if the ```ChangeManagement``` function has been called properly. Of course I expect that there is a test for ```ChangeManagment``` asserting that this function fires the command line as expected. In this case the log message needs to be asserted also in the corresponding test for ```ChangeManagement```.

Reply:
Yes, that should be tested in the ChangeManagementTest.class.

Reply:
Currently working on that. Check this branch: https://github.com/marcusholl/jenkins-library/tree/cm